pkg/controller/versions/strategy_v2_1.go

```diff
@@ -181,6 +182,17 @@ func (v *versionStrategyV2_1) validateMixerDisabled(spec *v2.ControlPlaneSpec, a
 	return allErrors
 }
 
+func (v *versionStrategyV2_1) validateAddons(spec *v2.ControlPlaneSpec, allErrors []error) []error {
+	if spec.Addons == nil {
+		return allErrors
+	}
+
+	if spec.Addons.ThreeScale != nil && spec.Addons.ThreeScale.Enabled != nil && *spec.Addons.ThreeScale.Enabled {
+		allErrors = append(allErrors, fmt.Errorf("support for 3scale has been removed in v2.1; please configure the 3scale WebAssembly adapter using a ServiceMeshExtension resource"))
```

Comment:
Should we also explicitly instruct the user to remove the 3scale block?

Reply:
+1

Reply:
Done.